### PR TITLE
HOCS-3572 Adding retry to backend calls

### PR DESCRIPTION
**Pull request by gardnersam**
This PR adds in axios-retry to retry any failed requests to the backend.

### CODE REVIEW

**Review by philbarton — COMMENTED** [review]
Looks gret to me. but maybe one of our JS gurus could have a look too.

File: server/libs/request.js [hunk 1]
```diff
@@ -17,12 +19,25 @@ const getHttpsClient = () => {
...
+    const logger = getLogger();
+
     const client = axios.create({
         baseURL,
         auth,
         httpsAgent: isProduction ? getHttpsClient() : null
     });
 
+    //Retry 3 times with 5 second gaps
+    axiosRetry(client, { retries: 3, retryDelay: () => { return 5000; } });
```
Comment (LiamMacP) [inline server/libs/request.js:31]: Worth having these configurable? If we have an issue in production, we can easily then turn this off?

File: server/config.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ const isProduction = process.env.NODE_ENV === 'production';
 const workflowAuth = (process.env.WORKFLOW_BASIC_AUTH || 'UNSET:UNSET').split(':');
 const isNotProd = process.env.IS_NOTPROD === '1';
 const showStackTraceInErrorPage = process.env.SHOW_STACKTRACE_ON_ERROR_PAGE === '1';
+const backendRequestRetries = process.env.BACKEND_REQUEST_RETRIES || 0;
```
Comment (zuzak) [inline server/config.js:5]: Set this to a sensible production default instead of 0.

File: server/libs/request.js [hunk 3]
```diff
@@ -17,12 +20,25 @@ const getHttpsClient = () => {
...
+    const logger = getLogger();
+
     const client = axios.create({
         baseURL,
         auth,
         httpsAgent: isProduction ? getHttpsClient() : null
     });
 
+    //Retry 3 times with 5 second gaps
+    axiosRetry(client, { retries: backendRequestRetries, retryDelay: () => { return 5000; } });
```
Comment (zuzak) [inline server/libs/request.js:32]: Should this be exponential backoff?

File: server/libs/request.js [hunk 4]
```diff
@@ -17,12 +20,25 @@ const getHttpsClient = () => {
...
+
+    //Warn about retries
+    client.interceptors.request.use(request => {
+        if('axios-retry' in request){
+            logger.warn('REQUESTRETRY', { url: request.url, retryCount: request['axios-retry'].retryCount });
```
Comment (zuzak) [inline server/libs/request.js:37]: Should this log the reason for the retry? Or the correlation ID of the initiating request?

**Review by fewtrees-ho — APPROVED** [review]

**Pull request merged**